rest/method_item_patch.go

```diff
@@ -2,20 +2,59 @@ package rest
 
 import (
 	"context"
+	"encoding/json"
+	"io/ioutil"
 	"net/http"
+	"strings"
 
+	jsonpatch "github.com/evanphx/json-patch"
 	"github.com/rs/rest-layer/resource"
 	"github.com/rs/rest-layer/schema/query"
 )
 
+func isJSONPatch(r *http.Request) bool {
+	if ct := r.Header.Get("Content-Type"); ct != "" && strings.TrimSpace(strings.SplitN(ct, ";", 2)[0]) == "application/json-patch+json" {
+		return true
+	}
+	return false
+}
+
+func isNoContent(r *http.Request) bool {
```

Comment:
Suggesting some changes, see comment for where it's called from in the code.

Reply:
Done